### PR TITLE
journal-importer: Consider ECONNRESET as EOF

### PR DESCRIPTION
**Pull request by daandemeyer**
Otherwise we log a noisy error when we get ECONNRESET.

### CODE REVIEW

File: src/shared/journal-importer.c [hunk 1]
```diff
@@ -93,6 +93,9 @@ static int get_line(JournalImporter *imp, char **line, size_t *size) {
                          imp->buf + imp->filled,
                          MALLOC_SIZEOF_SAFE(imp->buf) - imp->filled);
                 if (n < 0) {
+                        if (errno == ECONNRESET)
```
Comment (yuwata) [inline src/shared/journal-importer.c]: `ERRNO_IS_DISCONNECT(errno)` ?
Also, showing debugging log is better, I think.

File: src/shared/journal-importer.c [hunk 2]
```diff
@@ -134,6 +137,9 @@ static int fill_fixed_size(JournalImporter *imp, void **data, size_t size) {
                 n = read(imp->fd, imp->buf + imp->filled,
                          MALLOC_SIZEOF_SAFE(imp->buf) - imp->filled);
                 if (n < 0) {
+                        if (errno == ECONNRESET)
```
Comment (yuwata) [inline src/shared/journal-importer.c]: Also here.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Review by bluca — APPROVED** [review]

File: src/shared/journal-importer.c [hunk 3]
```diff
@@ -93,7 +93,12 @@ static int get_line(JournalImporter *imp, char **line, size_t *size) {
                          imp->buf + imp->filled,
                          MALLOC_SIZEOF_SAFE(imp->buf) - imp->filled);
                 if (n < 0) {
-                        if (errno != EAGAIN)
+                        if (ERRNO_IS_DISCONNECT(errno)) {
+                                log_debug("Got disconnect for importer %s", strna(imp->name));
```
Comment (YHNdnzj) [inline src/shared/journal-importer.c]: log_debug_errno to keep errno in structured log?

Comment (YHNdnzj) [reply src/shared/journal-importer.c]: Also, please add a period

File: src/shared/journal-importer.c [hunk 4]
```diff
@@ -134,7 +139,12 @@ static int fill_fixed_size(JournalImporter *imp, void **data, size_t size) {
                 n = read(imp->fd, imp->buf + imp->filled,
                          MALLOC_SIZEOF_SAFE(imp->buf) - imp->filled);
                 if (n < 0) {
-                        if (errno != EAGAIN)
+                        if (ERRNO_IS_DISCONNECT(errno)) {
+                                log_debug("Got disconnect for importer %s", strna(imp->name));
```
Comment (YHNdnzj) [inline src/shared/journal-importer.c]: ditto

**Review by YHNdnzj — APPROVED** [review]

**Review by yuwata — APPROVED** [review]

**Pull request merged**